src/utill.rs

```diff
@@ -81,11 +82,48 @@ pub fn setup_logger() {
                 .default_filter_or("coinswap=info")
                 .default_write_style_or("always"),
         )
-        .is_test(true)
+        // .is_test(true)
         .init();
     });
 }
 
+pub fn tor_instance_setup() {
+    Once::new().call_once(|| {
+        thread::spawn(|| {
+            let maker_file_path = PathBuf::from("/tmp/tor-rust/maker");
+            if !maker_file_path.exists() {
+                fs::create_dir_all(maker_file_path).unwrap();
+            }
+            let taker_file_path = PathBuf::from("/tmp/tor-rust/taker");
+            if !taker_file_path.exists() {
+                fs::create_dir_all(taker_file_path).unwrap();
+            }
+            let directory_file_path = PathBuf::from("/tmp/tor-rust/directory");
+            if !directory_file_path.exists() {
+                fs::create_dir_all(directory_file_path).unwrap();
+            }
+            
+            Tor::new()
+            .flag(TorFlag::DataDirectory("/tmp/tor-rust/".into()))
+            .flag(TorFlag::SocksPort(19050))
+            .flag(TorFlag::HiddenServiceDir("/tmp/tor-rust/maker/hs-dir".into()))
+            .flag(TorFlag::HiddenServiceVersion(HiddenServiceVersion::V3))
+            .flag(TorFlag::HiddenServicePort(TorAddress::Port(6102),None.into(),))
+            .flag(TorFlag::HiddenServicePort(TorAddress::Port(16102),None.into(),))
+            .flag(TorFlag::HiddenServicePort(TorAddress::Port(26102),None.into(),))
+            .flag(TorFlag::HiddenServicePort(TorAddress::Port(36102),None.into(),))
+            .flag(TorFlag::HiddenServicePort(TorAddress::Port(46102),None.into(),))
+            .flag(TorFlag::HiddenServiceDir("/tmp/tor-rust/taker/hs-dir".into()))
+            .flag(TorFlag::HiddenServiceVersion(HiddenServiceVersion::V3))
+            .flag(TorFlag::HiddenServicePort(TorAddress::Port(8000), None.into()))
+            .flag(TorFlag::HiddenServiceDir("/tmp/tor-rust/directory/hs-dir".into()))
+            .flag(TorFlag::HiddenServiceVersion(HiddenServiceVersion::V3))
+            .flag(TorFlag::HiddenServicePort(TorAddress::Port(8001), None.into()))
```

Comment:
We should ot bind to all these tor ports at the same time. Each port 6102, 16102 etc are allocated for different makers. So depending on test cases, not all of them should have a hidden service.